mathemagical-prototype.js

```diff
@@ -404,6 +413,76 @@ class Square {
   }
 }
 
+// TODO: refactor to use Vertex objects instead of p5 vectors
+class Line {
+  constructor(graphWindow, x1, y1, x2, y2) {
+    this.graphWindow = graphWindow;
+    this.verticesInGraph = [
+      createVector(x1, y1),
+      createVector(x2, y2)
+    ]
+    this.verticesInCanvas = this.getVerticiesInCanvas();
+    this.strokeWeight = graphWindow.getStrokeWeight();
+    this.isDraggable = false;
+  }
+
+  setIsDraggable(isDraggable) {
+    this.isDraggable = isDraggable;
+  }
+
+  getVerticiesInCanvas() {
+    return this.verticesInGraph.map(
+      v => createVector(this.graphWindow.X(v.x), this.graphWindow.Y(v.y))
+    );
+  }
+
+  computeGraphFromCanvas() {
```

Comment:
The map() method is more concise than the loop I used for the same method in the Square class. There are some trade-offs, but if one implementation is clearly preferable on balance, then could you go ahead and make the corresponding methods in these classes consistent? Once we get to the point where we have a third repetition of these methods, we may want to go ahead and abstract them into a base class, but for now, we could skip that.

Reply:
I don't know if we have to go with 100% loops or 100% functional-style .map() stuff.  For example, at my last job, we had the convention of using loops whenever the internal code mutated some state, and preferred functional-style for most other cases.  Sometimes, for more complicated cases, a loop will be much easier to deal with.

Reply:
But yeah, we can make it consistent between methods that do the same thing and/or combine them into common abstract methods.